.eleventy.js

```diff
@@ -157,46 +157,63 @@ module.exports = function (eleventyConfig) {
   eleventyConfig.addFilter("link", function (str) {
     return (
       str &&
-      str.replace(/\[\[(.*?\|.*?)\]\]/g, function (match, p1) {
+      str.replace(/\[\[(.*?\|?.*?)\]\]/g, function (match, p1) {
         //Check if it is an embedded excalidraw drawing or mathjax javascript
+        
         if (p1.indexOf("],[") > -1 || p1.indexOf('"$"') > -1) {
           return match;
         }
+        
         const [fileLink, linkTitle] = p1.split("|");
-
         let fileName = fileLink.replaceAll("&amp;", "&");
-        let header = "";
-        let headerLinkPath = "";
-        if (fileLink.includes("#")) {
-          [fileName, header] = fileLink.split("#");
-          headerLinkPath = `#${headerToId(header)}`;
-        }
-
-        let permalink = `/notes/${slugify(fileName)}`;
-        let noteIcon = process.env.NOTE_ICON_DEFAULT;
         const title = linkTitle ? linkTitle : fileName;
         let deadLink = false;
-
-        try {
-          const startPath = "./src/site/notes/";
-          const fullPath = fileName.endsWith(".md")
-            ? `${startPath}${fileName}`
-            : `${startPath}${fileName}.md`;
-          const file = fs.readFileSync(fullPath, "utf8");
-          const frontMatter = matter(file);
-          if (frontMatter.data.permalink) {
-            permalink = frontMatter.data.permalink;
-          }
-          if (frontMatter.data.tags && frontMatter.data.tags.indexOf("gardenEntry") != -1) {
-            permalink = "/";
-          }
-          if (frontMatter.data.noteIcon) {
-            noteIcon = frontMatter.data.noteIcon;
-          }
-        } catch {
-          deadLink = true;
+        let noteIcon = process.env.NOTE_ICON_DEFAULT;
+        let permalink;
+        let headerLinkPath;
+        
+          
+        if (fileLink.slice(-5).includes(".") && fileLink.slice(-3)!=".md"){
```

Comment:
Why does slice(-5).includes(".") imply its a file? This seems to only work with filetypes that have an extension that is smaller than 5 characters.